### PR TITLE
add InMemoryBackend 

### LINKED ISSUE #152: In-memory mosaic backend
If you have a mosaic in memory and you want to access backend methods like `.tile`, you currently need to know that you can pass `FileBackend(path=None, mosaic_def=...)`. It might be clearer to expose a `InMemoryBackend` that doesn't require a `path` argument.

### PR DESCRIPTION
**Pull request by vincentsarago**
closes #152

### CODE REVIEW

File: cogeo_mosaic/backends/memory.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    tms: TileMatrixSet = attr.ib(init=False, default=WEB_MERCATOR_TMS)
+
+    # default values for bounds and zoom
+    bounds: Tuple[float, float, float, float] = attr.ib(
+        init=False, default=(-180, -90, 180, 90)
+    )
+    minzoom: int = attr.ib(init=False, default=0)
+    maxzoom: int = attr.ib(init=False, default=30)
+
+    path: str = attr.ib(init=False, default=":memory:")
```
Comment (vincentsarago) [inline cogeo_mosaic/backends/memory.py:38]: `path` is outside the `__init__` and will always default to `:memory:`

File: cogeo_mosaic/backends/__init__.py [hunk 2]
```diff
@@ -16,8 +17,11 @@ def MosaicBackend(url: str, *args: Any, **kwargs: Any) -> BaseBackend:
     """Select mosaic backend for url."""
     parsed = urlparse(url)
 
+    if not url or url == ":memory:":
+        return InMemoryBackend(*args, **kwargs)
```
Comment (vincentsarago) [inline cogeo_mosaic/backends/__init__.py:21]: if url is `":memory:"` or set to None we use the InMemoryBackend

File: docs/advanced/backends.md [hunk 3]
```diff
@@ -47,6 +57,13 @@ with MosaicBackend("https://mosaic.com/amosaic.json.gz") as mosaic:
 
 with MosaicBackend("stac+https://my-stac.api/search", {"collections": ["satellite"]}, 10, 12) as mosaic:
     assert isinstance(mosaic, cogeo_mosaic.backends.stac.STACBackend)
+
+# In Memory (write)
```
Comment (kylebarron) [inline docs/advanced/backends.md:61]:
```suggestion
# In Memory (write)
# You can pass either None or ':memory:' to define an in-memory backend
```

File: docs/advanced/backends.md [hunk 4]
```diff
@@ -19,6 +19,16 @@ Read only backend won't allow `mosaic_def` in there `__init__` method. `.write()
 - **HTTP/HTTPS** (`http://`, `https://`)
 - **STAC** (`stac+:https://`). Based on [SpatioTemporal Asset Catalog](https://github.com/radiantearth/stac-spec) API.
 
+#### InMemory
+
+If you have a mosaicjson document and want to use the different backend methods you can use the spetial `InMemoryBackend`.
```
Comment (kylebarron) [inline docs/advanced/backends.md:24]:
```suggestion
If you have a mosaicjson document and want to use the different backend methods you can use the special `InMemoryBackend`.
```

File: docs/advanced/backends.md [hunk 5]
```diff
@@ -19,6 +19,16 @@ Read only backend won't allow `mosaic_def` in there `__init__` method. `.write()
 - **HTTP/HTTPS** (`http://`, `https://`)
 - **STAC** (`stac+:https://`). Based on [SpatioTemporal Asset Catalog](https://github.com/radiantearth/stac-spec) API.
 
+#### InMemory
```
Comment (kylebarron) [inline docs/advanced/backends.md:22]:
```suggestion
#### In-Memory
```

File: cogeo_mosaic/backends/memory.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+from morecantile import TileMatrixSet
+from rio_tiler.constants import WEB_MERCATOR_TMS
+from rio_tiler.io import BaseReader, COGReader
+
+from cogeo_mosaic.backends.base import BaseBackend, _convert_to_mosaicjson
+from cogeo_mosaic.mosaic import MosaicJSON
+
+
+@attr.s
+class InMemoryBackend(BaseBackend):
```
Comment (kylebarron) [inline cogeo_mosaic/backends/memory.py:15]: I think maybe `MemoryBackend` would be clearer than `InMemoryBackend`...

**Review by kylebarron — APPROVED** [review]

**Pull request merged**